### ISSUE TITLE
TypeScript Typings not fully exposed. 

### ISSUE THREAD

**Issue by Exigerr** [cure53/DOMPurify#367]
Greetings!

### Background & Context

I'm trying to make use of some of DOMPurify's TypeScript typings (`HookName`, `HookEvent` and `Config`) in properties and function/method parameters. With version 0.0.33 `@types/dompurify`, it seems that the interfaces and types are not usable on their own. I believe this is because the namespace and types are not fully exposed.

* DOMPurfiy version: 2.0.1
* @types/dompurify version: 0.0.33
* Typescript version: 3.5.3

#### Example

```typescript
//Simple example:
import DOMPurify from "DOMPurify";

function removeBlacklistedStyles(element: Element,  data: DOMPurify.SanitizeElementHookEvent, config: DOMPurify.Config) {
   //some node style attribute filtering...
}
DOMPurify.addHook("uponSanitizeAttribute", removeBlackListedStyles);
DOMPurify.sanitize(`<div style="color: #f00; border: 1px solid #fff; position: fixed;"></div>`);
```

The above will result in TypeScript compilation errors due to `removeBlacklistedStyles` parameter types containing `DOMPurify.SanitizeElementHookEvent` and `DOMPurify.Config`-  `TS2503: Cannot find namespace 'DOMPurify'`. 

As the types are not explicitly exported, the namespace is the only value we can work off.

### Feature/Bug proposed solution

I am not sure if this is the intended implementation or a bug/oversight. Believe this could be resolved by simply wrapping the types within a `declare namespace` code block:

```typescript
export as namespace DOMPurify;
export = DOMPurify;

declare const DOMPurify: DOMPurify.createDOMPurifyI;

declare namespace DOMPurify {
   interface createDOMPurifyI extends DOMPurifyI {
      //...
    }

   interface DOMPurifyI {
      //...
   }

   interface Config {
      //...
   }

   type HookName = //...

   type HookEvent = //...

   interface SanitizeElementHookEvent {
      //...
   }

   interface SanitizeAttributeHookEvent {
      //...
   }
}
```

If the above approach would be an acceptable solution, I'd be happy to make a PR for the changes.

Apologies if I should of raised this on the `DefinitelyTyped` repo instead!

**Comment by cure53** [comment]
Sounds great from our end :D

> Apologies if I should of raised this on the DefinitelyTyped repo instead!

I think probably yes.

**Comment by tomholub** [comment]
I suggest to call things `XxxInterface` rather than `XxxI`. But also, the proposal somewhat confuses me, maybe if I saw it as a PR, it would be easier to evaluate.

**Comment by Exigerr** [comment]
Made a PR for the changes and added in the typing the `.sanitize` method scenario in which it returns `TrustedHTML`.

@tomholub  
Could rename the interface if people are happy with it. Personally use the more traditional approach of `IThing`, but understand there are valid opinions against such naming conventions. At the moment, I have tried to keep things simple by maintaining the existing declarations but wrapped them in a namespace.
Apologies if the example was hard to follow. Hopefully the PR will better highlight things :)

**Comment by cure53** [comment]
Hey all, is it okay to close this ticket?

**Comment by Exigerr** [comment]
Aye, I think so.
The PR is in and awaiting final review and merge.
Thanks! 😄

**Comment by cure53** [comment]
Many thanks! :bowing_man:
